src/platform/imx8/include/platform/drivers/interrupt.h

```diff
@@ -40,9 +40,9 @@
 #define IRQ_MASK_IRQSTR_DSP6	(1 << IRQ_NUM_IRQSTR_DSP6)
 #define IRQ_MASK_IRQSTR_DSP7	(1 << IRQ_NUM_IRQSTR_DSP7)
 
-/* no nested interrupts */
-#define PLATFORM_IRQ_HW_NUM	0
-#define PLATFORM_IRQ_CHILDREN	0
+/* 32 HW interrupts + 8 IRQ_STEER lines each with 64 interrupts */
+#define PLATFORM_IRQ_HW_NUM	32
```

Comment:
The best solution would be to define it like this for every platform:
`#define PLATFORM_IRQ_HW_NUM       XCHAL_NUM_INTERRUPTS`

The question is, if it's not going to break things for other platforms.

Reply:
I wonder what the platforms without cascaded interrupts would do if somehow register_interrupt was called with an interrupt above `XCHAL_NUM_INTERRUPTS`. So not sure if I should do this now. Should I? Those with cascaded interrupts (apl, cnl, icl, sue) should be fine, not sure about byt. Would CI be able to capture issues on BYT if I did something like that?

Reply:
I think I may revert the change for `interrupt_is_dsp_direct` and indeed do the define with `XCHAL_NUM_INTERRUPTS`, so as to minimize any impact (otherwise those platforms would suddenly believe they have cascaded interrupts enabled). Will do.

Reply:
Then it's a clear bug in the code, but I think XTOS handles such cases. CI should catch this, but it's very unstable right now.

Reply:
Alright, pushed (IDK why Github didn't notice and it still shows the old commits?)

Reply:
Maybe you forgot about forcing?

Reply:
I pushed with -f because I edited the commits themselves. Not sure why it happens, the Git client does say "forced update".

Reply:
Finally, had to do some weird stuff to get it to recognize the force push. Waiting for new comments and for the SOFCI.

Reply:
Just noticed, using XCHAL_NUM_INTERRUPTS isn't 32 on any platform besides iMX. Would this shift of interrupts cause (possibly subtle) issues on such platforms (registering the wrong interrupt number)?

Reply:
That's why my original question. I'm not sure if cascaded interrupts are going to work correctly with this change. CI results suggest it doesn't, but result server is down, so I cannot check the exact failures.

Reply:
@paulstelian97 I think we should go with your first version where we keep the number of interrupts unchanged but we separate hw interrupts from the rest. This guarantees that the code will work as before, but we also allow irq steer implementation to move on.

It's a little bit unfortunate that cascading irq implementation doesn't take into account the real number of hardware interrupts.

Reply:
Maybe @lyakh knows how to do it right with this change, since he is the original author of the most of interrupt code.

Reply:
@tlauda Until his feedback comes in should I revert to the original version of this patch (which simply made the numbers equal on all platforms besides imx)? This should ensure nothing breaks due to this change. Right now with the usage of XCHAL_NUM_INTERRUPTS we get that unusual shift which may be troublesome (could affect any of the clients that uses an interrupt that goes through the cascading controller).